### PR TITLE
write some documentation and create a class with submodules

### PR DESCRIPTION
**Pull request by monsieurp**
* tb/backend.py: document functions
* tb/cli.py: create Tinderbox
* tinderbox.py: use Tinderbox class
* README.md: write documentation

### CODE REVIEW

File: tb/cli.py [hunk 1]
```diff
@@ -1,14 +1,16 @@
-"""the adjust tinderbox
+import logging
+import pydoc
```
Comment (alip) [inline tb/cli.py]: pydoc is unused.

File: tb/cli.py [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
-Options:
-    -h|--help     display help.
-    --version     display version.
-"""
+import tb.backend as tbb
+
+
+logging.basicConfig(
+    level=logging.INFO, format="[%(asctime)s][%(name)s][%(levelname)s]: %(message)s"
+)
```
Comment (alip) [inline tb/cli.py:10]: This configuration is done each time in every module. We could deduplicate it.

File: tb/cli.py [hunk 3]
```diff
@@ -18,28 +20,45 @@
...
-    tbb.run()
-    return 0
+        try:
+            runner[args["<command>"]]()
+        except KeyError:
+            sys.exit(
+                """Error! \'%s\' is an invalid command!
+See 'tinderbox --help' for a complete list of commands."""
+                % (args["<command>"])
+            )
```
Comment (alip) [inline tb/cli.py:58]: I think we could simplify this a bit, how about this?
```suggestion
        cmd = args["<command>"]
        if cmd not in runner:
            sys.exit(f"""Error! '{cmd}' is an invalid command!
See 'tinderbox --help' for a complete list of commands.""")
        runner[cmd]()
```

**Review by alip — APPROVED** [review]
lgtm in general, i have a few suggestions.

File: tb/cli.py (hunk 1 above)
Comment (monsieurp) [reply tb/cli.py]: thx updated.

File: tb/cli.py (hunk 3 above)
Comment (monsieurp) [reply tb/cli.py:58]: I prefer the try catch version 👀

File: tb/cli.py (hunk 2 above)
Comment (monsieurp) [reply tb/cli.py:10]: I've been looking for a way to refactor this code ... to no avail :(

**Pull request merged**